### DISCUSSION TITLE
WiFi + PSRAM == boom

### DISCUSSION THREAD

**Discussion by davepl** [PlummersSoftwareLLC/NightDriverStrip#308] [General]
For some reason, on the S3, if I enable PSRAM as the default for allocations early in setup, it crashes as soon as WiFi.begin connects.  It works on the WROVER, however.

So for now, I've disabled it except for MESMERIZER.  I also fixed a couple of bugs that would prevent strips longer than 255 from working right, which was a regression from a recent change.  Both will be merged into main shortly!

**Comment by Bun-Bun2** [comment]
I had a similar issue with PSRAM on the S2. ESP32-S2-WROVER would compile and upload fine but on running would hit a StoreProhibited error when calling psram_allocator<> in ledbuffer.h.

**Comment by robertlipe** [comment]
S3 has lots of "well known" (frustrating) issues about having several _different_ memory pools - and only some of them can be used for DMA. Some RAM is byte addressable and some is only word addressable. As such, there are alignment issues in the various pools that can byte you.

https://marc.merlins.org/linux/talks/ESP32_Memory/html/siframes.html (From personal communication, I know that Marc has struggled with this basic issue a LOT in his various projects like https://github.com/marcmerlin)

Do you have a stack trace available?  I'm not an Espressif guy, but I know there be dragons described in https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-guides/memory-types.html and even some bugs in Espressif's own libraries, like https://github.com/espressif/esp-idf/issues/10373 and confusion like https://github.com/espressif/esp-idf/issues/10281 is common.

I don't know that any of this DOES apply here (yet), I'm just throwing out some caution that there are a lot of dragons here to be aware of. Some have been domesticated and some are just there to be known about and reckoned with.
